### PR TITLE
Cleanup CodeFactor issues with null comparisions

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Please fast review to avoid merge conflicts because of many files changed.

Please review commit by commit - every commit contains only single kind of change so you can do very fast review.

This corrects about 1% of the CodeFactor issues.

The PR replace `null != var` with `var != null` and `null == var` with `var == null`

All changes is made in VS Code with Regex patterns - one pattern by commit:
```
'\((null) != (\w+)\)' -> '($2 != $1)'

'\((null) != (\w+.\w+)\)' -> '($2 != $1)'

'\((null) != (\w+) ' -> '($2 != $1 '

'(null) != (\w+),' -> '$2 != $1,'

'if \((null) != (\w+(.\w+)+)\)' -> 'if ($2 != $1)'

'if \((null) != (.+)\)' -> 'if ($2 != $1)'

Manual replacement 1

'if \((null) != (\w+.\w+)' -> 'if ($2 != $1'

'\b(null) != (\w+)\)' -> '$2 != $1)'

'\b(null) != (\w+.\w+)\)' -> '$2 != $1)'

'\b(null) != (\w+.\w+.\w+)\)' -> '$2 != $1)'

'\((null) != (\w+.\w+),' -> '($2 != $1,'

'\((null) != (\w+.\w+) ' -> '($2 != $1 '

'\((null) != (\w+.\w+\[\w+.\w+\])' -> '($2 != $1'

'\b(null) != (\w+.\w+)' -> $2 != $1

------------------------------------------

'\((null) == (\w+)\)' -> '($2 == $1)'

'\((null) == (\w+.\w+)\)' -> '($2 == $1)'

'\((null) == (\w+) ' -> '($2 == $1 '

'if \((null) == (\w+.\w+)' -> 'if ($2 == $1'

'\b(null) == (\w+)\)' -> '$2 == $1)'

'\b(null) == (\w+.\w+)\)' -> '$2 == $1)'

'\((null) == (\w+.\w+\["\w+"].\w+)' -> '($2 == $1'

'\((null) == (\w+.\w+\["\w+"])' -> '($2 == $1'

```

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by markekraus — DISMISSED** [review]

File: src/System.Management.Automation/engine/remoting/common/WireDataFormat/EncodeAndDecode.cs
Comment (iSazonov) [reply src/System.Management.Automation/engine/remoting/common/WireDataFormat/EncodeAndDecode.cs]: This is single manual change I did :-)
The commit is replaced.

**Comment by iSazonov** [comment]
If you remember a year ago, I offered to fix style in tests. Response team was negative - it is better to spend your time on more useful things. It was a convincing position. Then we continued to receive many-many comments on the style of tests _each_ PR. Finally everyone ran out of patience and we fixed our tests.
We have long endured trailing spaces in our code. At one point, our patience ran out and we fixed it.
Now we understand that spend a lot of useless efforts to style and formatting. So @TravisEz13  suggested the use of CodeFactor.
We must configure it before we can effectively use it. This is being discussed in #6930.
We disable obvious rules which we'll never fix like "one type per file" or "ordering elements" and more. 
_The remaining problems, we have to be correct (partially) to use CodeFactor effectively._
Should we disable current issue? I believe no because we already discussed this and we already require that use the _C# style_ in our code. If we leave bad samples we lose meaning use CodeFactor which reports problems a entire file rather than the latest changes.
About risk see my https://github.com/PowerShell/PowerShell/pull/6930#issuecomment-393015925

**Comment by lzybkr** [comment]
I sampled the report from CodeFactor when it was first enabled and wasn't impressed. Maybe after you refine the rules it will show some value, but I'm still skeptical.

As one data point for my skepticism - you touched 187 files in this PR, but CodeFactor claims there are 119 issues fixed and only 99 of the issues fixed are of the `Do not use comparison where value goes first`.

So if you found 88 or more places to fix that the tool didn't, humans will continue to find similar nits that the tool won't. Contributors will have one **additional** barrier to contributing in addition to the human nitpickers.

If the nitpicking isn't adding value, you could try asking those making the comments to try less nitpicking.

**Comment by iSazonov** [comment]
When I joined the project, the first three months I was amazed each time I got different types of style comments from different people. Why team did it? Obviously to get good code. What changed this time? Nothing. We will not be able to change our behavior if we do not take any steps.
This situation zugzwang. If we do not correct these issues we are doomed to many style comments. Fix all - it's not real. We need to find a compromise by configuring CodeFactor and fixing some issues.
I would be happy not to do this work. But I definitely envy CoreFX code looks like.

> As one data point for my skepticism - you touched 187 files in this PR, but CodeFactor claims there are 119 issues fixed and only 99 of the issues fixed are of the Do not use comparison where value goes first.

On CodeFactor site it is mentioned that they use their config and some other analysis methods - so we can get "weird" and "amazing" results. After we merge our config we can investigate problems if we find them.

**Comment by lzybkr** [comment]
My last comment on this specific issue - CoreFx is roughly as inconsistent as PowerShell, with about 6% of the comparisons using `null` on the LHS.

With some irony, if you fix this, we'd be less like CoreFx. :)

**Review by bergmeister — APPROVED** [review]

**Comment by iSazonov** [comment]
>As one data point for my skepticism - you touched 187 files in this PR, but CodeFactor claims there are 119 issues fixed and only 99 of the issues fixed 

I believe that this difference comes from the fact that some code is under `if UNIX` (and we have still `if !CORECLR`).

**Comment by dantraMSFT** [comment]
@iSazonov, I have to agree with @lzybkr, I see no benefit to making this change and the rule should simply be disabled.

**Comment by iSazonov** [comment]
Resolve merge conflicts.

**Comment by TravisEz13** [comment]
restarted Linux CI, I'll be OOF for a couple of days.

**Comment by daxian-dbw** [comment]
@TravisEz13 @adityapatwardhan and I discussed this offline and we agree to take this PR for the same reasons as @SteveL-MSFT mentioned above. But in general, we prefer less changes per PR for style related changes:
- A PR is better to focus on a single category of style issue.
- A PR is better to modify less than 10 files (it also depends on how many changes per file)

I will soon go through this PR and merge it.

**Comment by iSazonov** [comment]
> •A PR is better to modify less than 10 files (it also depends on how many changes per file)

I suggest to apply this rule only to code. We have tons style issues for doc comments - I think we could change more files per PR in the case without risk.

Seems also the rule assumes that we could fix style issues by file, by module, by cmdlet set (ex., web cmdlets), by provider and so on.

Does this conclusion imply that our policy is changed and we want and accept style changes?
I believe we should update our internal contributing docs.

**Comment by daxian-dbw** [comment]
@iSazonov Doc comments issues can be an exception, because of the super low risk. It's not a rule, but a flexible guideline for us maintainers to consider (you can read the tone -- `prefer`, `better to` :)).

The main reason that we don't want massive changes in a PR is the time cost to review -- it will be easy for the PR to end up with no approval since no one wants to spend the time reviewing all the changes. This will, in turn, hurt the author's feeling and discourage the author's further contribution, because if the author really meant to fix those style issues, he/she must have put a lot effort/time into it. So we want to set some guidelines to help prevent a contributor from getting in that situation.

**Comment by daxian-dbw** [comment]
> Seems also the rule assumes that we could fix style issues by file, by module, by cmdlet set (ex., web cmdlets), by provider and so on.
>
> Does this conclusion imply that our policy is changed and we want and accept style changes?
I believe we should update our internal contributing docs.

We accept PRs that are purely for style changes (we have to if we want to make CodeFactor useful). Such a PR should be reviewable, and that's why we proposed the flexible guideline. But this should be temporary -- once the existing style issues reported by CodeFactor are fixed, we won't accept PR with pure style changes.

**Review by daxian-dbw — APPROVED** [review]
Went through the changes. LGTM.

**Comment by iSazonov** [comment]
@daxian-dbw Thanks for code review and clarifications!

**Pull request merged**
